docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,557 @@
+
+In order to have `constructor` cover our needs, we need to add the features ourselves. Upstream
+maintenance is meant to be improve over the year, but for now the reviews are coming in slow. As
+a result, we are temporarily forking the project and developing the features as needed while
+submitting PRs to upstream [^constructor-upstream] to keep things tidy. Our improved `constructor`
+fork has the following features:
+
+* Cross-platform shortcut creation for the distributed application thanks to a complete `menuinst`
+  [^menuinst] rewrite
+* PKG installers on macOS can be customized, signed and notarized
+* `constructor` can deploy more than one conda environment at once now
+* Extra files can be added to the installation without having to use an extra conda package
+* Improvements for license collection and analysis
+* Small fixes for reported installation size, progress reports and other cosmetic details
+
+The resulting product now is able to install napari across operating systems with fully working
+shortcuts that respect the activation mechanism of `conda` environments to guarantee that all
+dependencies work properly. Having `constructor` distribute multiple environments at once allow us
+to split the installation like this:
+
+* A `base` environment with `conda`, `mamba` and `pip` to manage the other `conda` environments.
+* A `napari-X.Y.Z` environment (X.Y.Z being the version of the bundled release) with napari and
+  its dependencies.
+
+This separation allows us to:
+
+* Handle napari updates by simply creating a fresh environment with the new version (more on this
+  in Milestone 4).
+* Apply reparations in the environment without worrying that a faulty plugin install can render
+  the whole environment non-functional.
+* If necessary, adding support for "napari projects": a napari installation with a specific
+  combination of plugins. Useful in the event that a user wants to use plugins that cannot be
+  installed in the same environment due to conflicting dependencies (more details on this in
+  Milestone 3).
+
+The installer relies on conda-forge to obtain the needed packages. Pre-release installers
+can be built thanks to the nightlies available on the napari channel.
+
+### Milestone 3: Adding support for conda packages in the plugin manager
+
+napari has its own plugin manager, which so far relied on `pip` to install packages available on
+PyPI. To make it compatible with conda packaging, three key changes are needed:
+
+1.  The list of packages on conda-forge does not necessarily match the one coming from PyPI. Right
+    now, the plugins on conda-forge are a _subset_ of those on PyPI, but this might change in the
+    future if some napari plugins become available on conda-forge but not PyPI due to packaging
+    limitations (e.g. availability of dependencies). As a result, the plugin manager needs to source
+    the list of plugins from a repository-agnostic source: the napari hub API. It must be noted that
```

Comment:
We have the same concern for PyPI, no?

Reply:
I would rephrase the Q to be:

What kinds of guarantees does the napari hub team need to commit to to give the napari team confidence in using the napari hub API as the primary registry of plugins in the napari ecosystem?

With the napari hub API (unlike scraping PyPI HTML), we have more control to specify what kind of uptime we expect and the napari hub team can commit to whatever level of uptime the napari community needs, monitor it, and put in the necessary "on call" mechanisms we need to support it.

Reply:
Is it feasible to push the JSON payload returned by the napari hub API to a github repo every hour or so? That way we can use that as a fallback in case the API is down or some other issue.